src/esp/bindings_js/navigate.js

```diff
@@ -56,42 +55,38 @@ class NavigateTask {
     this.components.status.innerHTML = text;
   }
 
-  applyGamma(data, gamma) {
-    for (let i = 0; i < data.length; i++) {
-      data[i] = Math.pow(data[i]/255.0, gamma) * 255;
-    }
-  }
-
   renderImage() {
-    const obs = this.sim.getObservation("rgb", null);
-    this.imageData.data.set(obs.getData());
-    // convert from linear to sRGB gamma
-    this.applyGamma(this.imageData.data, 2.2);
-    this.imageCtx.putImageData(this.imageData, 0, 0);
+    // Not using this as of now, but useful for later tracking purposes
+    // Also, required to render everything on sim side
+    this.lastObs = this.sim.getObservation("rgb", null);
     this.renderRadar();
   }
 
   renderSemanticImage() {
+    if (this.semanticObjects.size() == 0) {
+      return;
+    }
+
     const obs = this.sim.getObservation("semantic", null);
     this.semantic_data = obs.getData();
     let data = this.semantic_data;
 
     // TOOD(msb) implement a better colorization scheme
     for (let i = 0; i < 640*480; i++) {
       if (data[i*4] & 1) {
-	this.semanticImageData.data[i*4] = 255;
+        this.semanticImageData.data[i*4] = 255;
```

Comment:
If you guys want, `clang-format` will format JS code!  Just update the regex here: https://github.com/facebookresearch/habitat-sim/blob/master/.circleci/config.yml#L48 and here: https://github.com/facebookresearch/habitat-sim/blob/master/.pre-commit-config.yaml#L30